subsys/bluetooth/mesh/lightness_srv.c

```diff
@@ -0,0 +1,805 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <stdlib.h>
+#include <bluetooth/mesh/lightness_srv.h>
+#include "model_utils.h"
+#include "lightness_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#define LOG_MODULE_NAME bt_mesh_light_srv
+#include "common/log.h"
+
+#define LVL_TO_LIGHT(_lvl) (repr_to_light((_lvl) + 32768, ACTUAL))
+#define LIGHT_TO_LVL(_light) (light_to_repr((_light), ACTUAL) - 32768)
+
+/** Persistent storage handling */
+struct bt_mesh_lightness_srv_settings_data {
+	struct bt_mesh_lightness_range range;
+	u16_t default_light;
+	u16_t last;
+	bool is_on;
+} __packed;
+
+#ifdef BT_DBG_ENABLED
+static const char *const repr_str[] = { "Actual", "Linear" };
+#endif
+
+static int store_state(struct bt_mesh_lightness_srv *srv)
+{
+	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		return 0;
+	}
+
+	struct bt_mesh_lightness_srv_settings_data data = {
```

Comment:
This might be unreachable code after always true condition.
better:
``` C
	#ifdef CONFIG_BT_SETTINGS
		struct bt_mesh_lightness_srv_settings_data data = {
		...
		return bt_mesh_model_data_store(srv->lightness_model, false, &data,
					sizeof(data));
	#else
		return 0;
	#endif
```

Reply:
@nvlsianpu That is intended. The compiler will see the always true / false condition, and remove the unused code. So it should end up the same. The advantage is that the compiler will still see the code and give error if it finds it, so we can avoid code-rot, or unnecessary "defined but not used" warnings.